### PR TITLE
Addition of NIFI guide.

### PR DESCRIPTION
**Pull request by lsac**
Hi Mark, please verify the NiFi guide. If there is errors, please revert back to me.

Thanks,
Wayne

### CODE REVIEW

**Comment by mdspielman** [comment]
I find it odd to have the samples here as: "solace-jndi" in a subdirectory. Why not use the Solace Developer samples or at least name the directory as to include "samples" in the name.

**Comment by mdspielman** [comment]
You need to add the new guide to the "guides.yml" in order for it to be added to the TOC.

https://github.com/SolaceLabs/solace-integration-guides/blob/ebc1793328d1a81e64d30a1a4b6d4d0f1f198083/docs/_data/guides.yml

File: docs/_guides/nifi-jms-jndi.md [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+* [Solace JMS API Online Reference Documentation]({{ site.links-docs-jms-api }}){:target="_top"}
+* [Solace Feature Guide]({{ site.links-docs-features }}){:target="_top"}
+* [Solace Message Router Configuration]({{ site.links-docs-router-config }}){:target="_top"}
+* [Solace Command Line Interface Reference]({{ site.links-docs-cli }}){:target="_top"}
+* [NiFi Documentation](http://nifi.apache.org/docs.html){:target="_blank"}
+* [NiFi Developer Guide](http://nifi.apache.org/developer-guide.html){:target="_blank"}
+
+## Integrating Solace with NiFi JMS Processor
+
+NiFi JMS Bundle comes with 2 processors, ConsumeJMS and PublishJMS. The 2 out-of-box processors work well with some messaging brokers, i.e. ActiveMQ and etc. However, not all messaging brokers work well with the built-in JMS processors that expect zero-argument connection factory constructors. Solace connection factory and some other JMS connection factories cannot be instantiated with zero argument. There is a current NiFi Jira on [this major bug](https://issues.apache.org/jira/browse/NIFI-2701){:target="_blank"}.
```
Comment (mdspielman) [inline docs/_guides/nifi-jms-jndi.md]: Solace has a way to work with zero-argument connection factor constructors:

SolConnectionFactory cf = SolJmsUtility.createConnectionFactory();
cf.setHost((String) args[0]);
cf.setVPN("default");
cf.setUsername("clientUsername");
cf.setPassword("password");

From http://dev.solace.com/get-started/jms-tutorials/publish-subscribe_jms/

Could that be used to help out?

File: docs/_guides/nifi-jms-jndi.md [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+2.	If you are new to Solace or your company does not have development message routers, you can obtain a trail Solace Virtual Message Router (VMR) from the [Solace Developer Portal Downloads]({{ site.links-downloads }}){:target="_top"}. For help getting started with your Solace VMR you can refer to [Solace VMR Getting Started Guides]({{ site.links-vmr-getstarted }}){:target="_top"}.
+
+The Solace JMS jars are required.  They can be obtained on [Solace Developer Portal Downloads]({{ site.links-downloads }}){:target="_top"} or from [Maven Central]({{ site.links-jms-maven }}){:target="_blank"}.
+
+##### Solace JMS Dependencies
+
+The easiest way to integrate Solace and NiFi is using the client libraries available via public [Maven Repositories](https://mvnrepository.com/artifact/com.solacesystems){:target="_blank"}. You can downloaded the libraries into a directory and reference them directly, the following resources are all required:
```
Comment (mdspielman) [inline docs/_guides/nifi-jms-jndi.md]: Better to use this link: [Maven Central]({{ site.links-jms-maven }}){:target="_blank"}

You did that two lines above as an example.

**Comment by mdspielman** [comment]
If you could address the comment on the link and the fix to guides.yml, then I would be to merge this pull request and you can treat the other feedback as suggestions.

**Comment by lsac** [comment]
Got it Mark. 

On May 18, 2017, at 4:35 PM, Mark Spielman <notifications@github.com> wrote:

If you could address the comment on the link and the fix to guides.yml, then I would be to merge this pull request and you can treat the other feedback as suggestions.

File: docs/_guides/nifi-jms-jndi.md (hunk 1 above)
Comment (lsac) [inline docs/_guides/nifi-jms-jndi.md]: SolJmsUtility requires package import, tight coupling and dependencies on Solace libraries.

File: docs/_guides/nifi-jms-jndi.md (hunk 2 above)
Comment (lsac) [inline docs/_guides/nifi-jms-jndi.md]: FIxed

**Comment by lsac** [comment]
All have been taken care of.

**Comment by mdspielman** [comment]
Thank you. Merged.

**Pull request merged**